rotkehlchen/db/utils.py

```diff
@@ -363,3 +365,67 @@ def combine_asset_balances(balances: List[SingleDBAssetBalance]) -> List[SingleD
             new_balances = _append_or_combine(new_balances, balance)
 
     return new_balances
+
+
+def db_schema_sanity_check(
+        db_name: Literal['private', 'global'],
+        cursor: 'DBCursor',
+        script_create_tables: str,
+) -> None:
+    """Ensures that database schema is not broken. Raises DBSchemaError if anything is bad."""
+    # Replacing new lines and white spaces since they may vary if by an accident code of a
+    # db upgrade was a bit different from the one that creates new tables
+    create_script_from_constant = script_create_tables.replace(' ', '').replace('\n', '').replace('\'', '"')  # noqa: E501
```

Comment:
This doesn't check for views. At the moment we don't have any in the schema but we have had. You should also check for that.

Reply:
Is it really needed? Maybe is better to add a check for views when we will have views?

Afaik there are many other different constructions / structures in sql and sqlite, so checking all of them when we don't have them sounds irrational to me.

What do you think?

Reply:
Yes that it is what I meant. At the moment we don't have any but maybe we should have a vacuum or something that drop it if exists and we don't have it. It happened to kelsos for example that he had an error because he had a view that shouldn't exists and this was breaking the upgrade